### DISCUSSION TITLE
When you skipRender in a parent component, child components get initialized (reset).

### DISCUSSION THREAD

**Discussion by nshiro** [livewire/livewire#4447] [7. Bugs (archived)] [answered]
I'm experiencing two things here.
When you use 'skipRender' in a parent component,

(1) Child components get initialized.
(2) You get an error message in console like 'Livewire: Multiple root elements detected. This is not supported'.

I won't discuss (2) here, because there are already some discussions about it.
I'll discuss (1) here.

## Steps to reproduce

php artisan make:livewire Boss    // This is a parent component. because 'Parent' was a reserved word.
php artisan make:livewire Child

Boss.php
```php
class Boss extends Component
{
    public function skip()
    {
        $this->skipRender();
    }

    public function noskip()
    {
    }

    public function render()
    {
        return view('livewire.boss');
    }
}
```

boss.blade.php
```php
<div>
    <livewire:child>

    <input type="button" value="Skip" wire:click="skip()">
    <input type="button" value="No Skip" wire:click="noskip()">
</div>

```

Child.php
```php
class Child extends Component
{
    public $counter = 0;

    public function increment()
    {
        $this->counter++;
    }

    public function render()
    {
        return view('livewire.child');
    }
}
```

child.blade.php
```php
<div>
    <h1>{{ $counter }}</h1>
    <input type="button" value="increment" wire:click="increment">
</div>
```

First you press 'increment' three times then you see '3'.
Next you press 'Skip' then press 'No Skip'.

Now you see '0'.
(Also, you see 'Livewire: Multiple root elements detected' message in console.)

![2021-12-23_22h20_10](https://user-images.githubusercontent.com/14008307/147248225-9ba87c4f-33e2-4385-b47e-37bb298f490b.png)

![2021-12-23_22h22_26](https://user-images.githubusercontent.com/14008307/147248268-89903114-acc7-47f4-baea-75a0c287f7d1.png)

I feel this is more like a bug if not designed so.

My Guess.
When you skipRender in a parent component, your PreviouslyRenderedChildren gets empty array.
So next time you don't skip render, your child components get initialized.

I don't know how to solve this. Maybe I should stop using skipRender ?

**Comment by joshhanley** [comment]
@nshiro hmm yeah it shouldn't be loosing track of the children! Can you explain what you are using skipRender for/ hoping to achieve?

Thanks!

**Reply by nshiro** [reply]
@joshhanley 
Well, I was using skipRender because I wanted to skip database access. I just wanted to toggle the modal show flag.
It was not for a real application. I was just playing around.
I think I now know how to fix it. Let me prepare for it.

Thank you.

**Comment by joshhanley** [comment, marked as answer]
@nshiro ah right ok yeah.

Personally I would just use Alpine to hide/ show the modal, that way you don't have any network requests, let alone DB calls.

But yeah it's not right that skip render breaks child components.

**Reply by nshiro** [reply]
@joshhanley 
Yes. I understand that.

Well, I wrote it incorrectly.
I wanted to toggle a modal flag and send an user id with it.

This is a simplified example in case you are interested.

ShowUsers.php
```php
    public $showDeleteModal = false;
    public $deleteUserId;

    public function showDeleteModal(int $id)
    {
        $this->showDeleteModal = true;
        $this->deleteUserId = $id;
        $this->skipRender();
    }

    public function deleteUser()
    {
        User::findOrFail($this->deleteUserId)->delete();

        $this->showDeleteModal = false;
    }
```

show-users.blade.php
```html
    <table>
        @foreach($users as $user)
        <tr>
            <td wire:click="showDeleteModal({{ $user->id }})">DELETE</td>
        </tr>
        @endforeach
    </table>


    <x-confirmation wire:model="showDeleteModal">
        <x-slot name="title">
            Are you sure?
        </x-slot>
        <x-slot name="content">
            <button wire:click="deleteUser()">Yes</button>
            <button wire:click="$set('showDeleteModal', false)">No</button>
        </x-slot>
    </x-confirmation>
```

Anyway ,I' ll make an PR.
Thank you.
